File: HackerRank/Algorithms/2. Implementation/Minimum_distances.java
import java.util.Scanner;
import java.lang.Math;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int A[] = new int[n];
        for(int A_i=0; A_i < n; A_i++){
            A[A_i] = in.nextInt();
        }

        int i, j, minimum = Integer.MAX_VALUE;
        for(i = 0; i < n; i++) {
          for(j = i + 1; j < n; j++) {
            if (A[i] == A[j]) {
              minimum = Math.min(minimum, Math.abs(i - j));
            }
          }
        }

        if (minimum == Integer.MAX_VALUE) {
          System.out.println(-1);
        } else {
          System.out.println(minimum);
        }
    }
}

File: HackerRank/Algorithms/2. Implementation/Minimum_distances.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <limits.h>

int inline min(int a, int b) {
  return (a < b ? a : b);
}

int main(){
    int n, i, j, minimum;
    scanf("%d",&n);
    int *A = malloc(sizeof(int) * n);
    for(i = 0; i < n; i++){
       scanf("%d",&A[i]);
    }

    //Logic
    minimum = INT_MAX;
    for(i = 0; i < n; i++) {
      for(j = i + 1; j < n; j++) {
        if (A[i] == A[j]) {
          minimum = min(minimum, abs(i - j));
        }
      }
    }
    free(A);

    if (minimum == INT_MAX) {
      printf("%d\n", -1);
    } else {
      printf("%d\n", minimum);
    }

    return 0;
}
